src/control-plane/backend/lambda/api/service/segment.ts

```diff
@@ -0,0 +1,337 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import {
+  CLICKSTREAM_SEGMENTS_CRON_JOB_RULE_PREFIX,
+  OUTPUT_USER_SEGMENTS_WORKFLOW_ARN_SUFFIX,
+  Segment,
+  SegmentDdbItem,
+  SegmentJobStatus,
+  SegmentJobTriggerType,
+} from '@aws/clickstream-base-lib';
+import {
+  DeleteRuleCommand,
+  EventBridgeClient,
+  PutRuleCommand,
+  PutTargetsCommand,
+  RemoveTargetsCommand,
+} from '@aws-sdk/client-eventbridge';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { SFNClient, StartExecutionCommand } from '@aws-sdk/client-sfn';
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner';
+import { NextFunction, Request, Response } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import { PipelineStackType } from '../common/model-ln';
+import { logger } from '../common/powertools';
+import { aws_sdk_client_common_config } from '../common/sdk-client-config-ln';
+import { ApiFail, ApiSuccess, BucketPrefix } from '../common/types';
+import { getBucketPrefix } from '../common/utils';
+import { CPipeline } from '../model/pipeline';
+import { DynamoDBSegmentStore } from '../store/dynamodb/dynamodb-segment-store';
+import { DynamoDbStore } from '../store/dynamodb/dynamodb-store';
+
+const segmentStore = new DynamoDBSegmentStore();
+const pipelineStore = new DynamoDbStore();
+const eventBridgeClient = new EventBridgeClient({
+  ...aws_sdk_client_common_config,
+  region: process.env.AWS_REGION,
+});
+const sfnClient = new SFNClient({
```

Comment:
I remember it's a resource in the pipeline stack; we should defer initiating this client based on the region of the given pipeline.

Reply:
Move SFNClient and EventBridgeClient initiation to APIs.